Sources/libcoverage/coverage.c

```diff
@@ -19,11 +19,17 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#if !defined(_WIN32)
 #include <sys/mman.h>
+#endif
 #include <sys/stat.h>
+#if !defined(_WIN32)
 #include <sys/time.h>
+#endif
 #include <sys/types.h>
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
```

Comment:
Why is this not `#if !defined(_WIN32)` as above? Maybe add a short comment since I'm sure I'll ask myself that question again the next time I see this code =)

Reply:
Ah, so, this is purely pedantry :-D.  `unistd.h` is the *Unix* standard header.  That only makes sense to import on Unix platforms.  The identification for such platforms is ... `__unix__`.  The thing is, MacOS X and new wants to be treated as a unix but doesn't declare itself to be unix based.  So, we use `__APPLE__` to identify if we are targeting an Apple platform, and if it is using mach,, we assume that it is a Unix-like environment instead of older MacOS platforms.  Technically, I suppose that A/UX and Rhapsody would also support `unistd.h` but the latter never saw release (and thus I cannot verify if it ever declared to be unix-esque) and the former has been discontinued.  In theory, if someone wanted to port this to an alternative operating system (e.g. SerenityOS) this ensures that this is closer to correct.

Reply:
Added a brief version of this explanation as well.

Reply:
Haha that's great, thanks for the explanation and comment! :D